instrumentation/rocketmq-client-4.8/testing/src/main/groovy/io/opentelemetery/instrumentation/rocketmq/AbstractRocketMqClientTest.groovy

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetery.instrumentation.rocketmq
+
+import base.BaseConf
+import io.opentelemetry.instrumentation.test.InstrumentationSpecification
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes
+import org.apache.rocketmq.client.producer.DefaultMQProducer
+import org.apache.rocketmq.client.producer.SendCallback
+import org.apache.rocketmq.client.producer.SendResult
+import org.apache.rocketmq.common.message.Message
+import org.apache.rocketmq.remoting.common.RemotingHelper
+import org.apache.rocketmq.test.client.rmq.RMQNormalConsumer
+import org.apache.rocketmq.test.client.rmq.RMQNormalProducer
+import org.apache.rocketmq.test.factory.ProducerFactory
+import org.apache.rocketmq.test.listener.rmq.concurrent.RMQNormalListener
+import org.apache.rocketmq.test.listener.rmq.order.RMQOrderListener
+import spock.lang.Shared
+import spock.lang.Unroll
+import static io.opentelemetry.api.trace.SpanKind.CONSUMER
+import static io.opentelemetry.api.trace.SpanKind.PRODUCER
+import static io.opentelemetry.instrumentation.test.utils.TraceUtils.basicSpan
+import static io.opentelemetry.instrumentation.test.utils.TraceUtils.runUnderTrace
+
+@Unroll
+abstract class AbstractRocketMqClientTest extends InstrumentationSpecification{
+
+  @Shared
+  private RMQNormalConsumer consumer
+
+  @Shared
+  private RMQNormalProducer producer
+
+  @Shared
+  DefaultMQProducer defaultMQProducer
+
+  @Shared
+  private String sharedTopic
+
+  @Shared
+  private String brokerAddr
+
+  @Shared
+  Message msg
+
+  @Shared
+  int consumeTime = 1000
+
+  @Shared
+  BaseConf baseConf =new BaseConf()
```

Comment:
Nit formatting: can you add a space after `=` everywhere where it's missing? Thanks